lib/puppet/network/http/response.rb

```diff
@@ -5,7 +5,23 @@ def initialize(handler, response)
   end
 
   def respond_with(code, type, body)
-    @handler.set_content_type(@response, type)
+    format = Puppet::Network::FormatHandler.format_for(type)
+    mime = format.mime
+    charset = format.charset
+
+    if charset && body.is_a?(String)
+      # REMIND: not all charsets are valid ruby encodings, e.g. ISO-2022-KR
+      encoding = Encoding.find(charset)
+
+      if body.encoding != encoding
+        # REMIND this can raise if body contains invalid UTF-8
+        body.encode!(encoding)
```

Comment:
@Iristyle @MosesMendoza I wanted to sync up with tomorrow about this.

1. Can we assume if we're using `application/json` or `text/plain` that the response body is always valid UTF-8 or can be at least be transcoded to it?
2. I'm assuming charset is really only going to be `utf-8` or `nil`. Does that seem reasonable?